cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -366,10 +360,10 @@ if (ARROW_BOOST_VENDORED)
     "${BOOST_LIB_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}boost_filesystem${CMAKE_STATIC_LIBRARY_SUFFIX}")
   set(BOOST_STATIC_REGEX_LIBRARY
     "${BOOST_LIB_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}boost_regex${CMAKE_STATIC_LIBRARY_SUFFIX}")
-  set(BOOST_SYSTEM_LIBRARY boost_system_static)
-  set(BOOST_FILESYSTEM_LIBRARY boost_filesystem_static)
-  set(BOOST_REGEX_LIBRARY boost_regex_static)
-  if (ARROW_BOOST_HEADER_ONLY)
+  set(BOOST_SYSTEM_LIBRARY boost_system)
```

Comment:
Can you explain why you're removing the "static" and "shared" prefixes everywhere? We're explicitly selecting one or the other variant of a library (for example, here we're linking statically with boost, while later below we're linking dynamically with it).

Reply:
I understand that sub-projects may choose different linkage type (static or shared) for a library.
I'll submit another solution.